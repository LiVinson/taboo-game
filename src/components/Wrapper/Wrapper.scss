.wrapper {
    background-image: linear-gradient(to bottom, var(--color-white) 0%, var(--color-white) 50%,
         70%, var(--color-primary) 90%);
    padding: 1rem 0rem 6rem 0rem;
    min-height:100vh;
    position: relative;
    overflow:hidden; 

}