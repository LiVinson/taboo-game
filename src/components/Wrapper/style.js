import styled from "styled-components"

export const StyledWrapper = styled.div`
  padding: 1rem 0rem 6rem 0rem;
  min-height: 100vh;
  position: relative;
  /* Needs z-index so contents inside will be aboe layered taboo cards */
  z-index:0;
  width: 30rem;
  max-width:100%;
  margin: 0 auto;  
`
