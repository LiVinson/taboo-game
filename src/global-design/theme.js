const theme = {
  color: {
    primary: "#1982C4", //light blue
    accent1: "#8AC926", //lime green
    accent2: "#ee5b2c", //orange
    accent3: "#a23b72", //purple
    primaryLight: "#58b1e9", //light-blue
    white: "#fff",
    offWhite: "#f7f9f9",
    lightGray1: "#f1f1f1",
    lightGray2: "#ebebeb",
    grayDark1: "#757575",
    grayDark2: "#333333"
  },
  text: {
    display: "'Oswald', sans-serif",
  },
  fontSize: {
    xsmall: "1.8rem",
    small: "2.2rem",
    medium: "2.7rem",
    large: "4rem",
    xlarge: "7rem",
  },
}


export default theme
