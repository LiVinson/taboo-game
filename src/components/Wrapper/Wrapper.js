import React from "react"
import PropTypes from "prop-types"
import { StyledWrapper } from "./style"

const Wrapper = ({ children }) => {
    return (
        <StyledWrapper>
            {children}
        </StyledWrapper>
    )
}

Wrapper.propTypes = {
    children: PropTypes.node.isRequired
}

export default Wrapper