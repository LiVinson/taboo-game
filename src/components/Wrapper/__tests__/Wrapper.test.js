import React from "react"
import { shallow } from "enzyme"
import renderer from "react-test-renderer"
import "jest-styled-components"
import { ThemeProvider } from "styled-components"
import theme from "../../../global-design/theme"
import Wrapper from "../Wrapper"

test("Wrapper renders correctly", () => {
  const children = <h1>I am a child element</h1>

  const wrapper = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Wrapper>
          {children}
        </Wrapper>
      </ThemeProvider>
    )
    .toJSON()

  expect(wrapper).toMatchSnapshot()
})

test("Wrapper renders props.children", () => {
  const children = <h1>I am a child element</h1>
  //Pass prop directly instead of wrapping with ThemeProvider, due to shallow rendering
  const WrapperComponent = shallow(<Wrapper theme={theme}>{children}</Wrapper>)
  expect(WrapperComponent.contains(children)).toBe(true)
})
