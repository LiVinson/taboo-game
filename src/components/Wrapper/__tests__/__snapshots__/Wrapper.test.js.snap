// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper renders correctly 1`] = `
.c0 {
  padding: 1rem 0rem 6rem 0rem;
  min-height: 100vh;
  position: relative;
  z-index: 0;
  width: 30rem;
  max-width: 100%;
  margin: 0 auto;
}

<div
  className="c0"
>
  <h1>
    I am a child element
  </h1>
</div>
`;
